specifyweb/frontend/js_src/lib/components/QueryBuilder/reducer.ts

```diff
@@ -156,11 +156,6 @@ export const reducer = generateReducer<MainState, Actions>({
       fields: replaceItem(state.fields, line, {
         ...state.fields[line],
         mappingPath: newMappingPath,
-        dataObjFormatter:
-          mappingPathIsComplete(newMappingPath) &&
-          action.currentTableName === action.newTableName
-            ? undefined
-            : state.fields[line].dataObjFormatter,
```

Comment:
Take a close look at this change @specify/dev-testing!
In my local testing, everything seems to work okay with this change, but I've yet to go deeper through the code and fully assess the consequences of the change.